lib/bamboo/adapters/smtp_adapter.ex

```diff
@@ -286,6 +286,9 @@ defmodule Bamboo.SMTPAdapter do
     Enum.reduce(config, [], &to_gen_smtp_server_config/2)
   end
 
+  defp to_gen_smtp_server_config({:server, {:system, var}}, config) do
+    [{:relay, System.get_env(var)} | config]
```

Comment:
While you're at changing the code, what do you think of re-calling the function using the interpolated value? It could make it less error-prone if we don't duplicate the configuration keys.

Example (always easier 😸 )
```elixir
defp to_gen_smtp_server_config({:server, {:system, var}}, config) do
  to_gen_smtp_server_config({:server, System.get_env(var)}, config)
end
```